riverdriver/river_driver_interface.go

```diff
@@ -36,11 +34,6 @@ type Driver[TTx any] interface {
 	// where multiple drivers are supported and is subject to change.
 	GetDBPool() *pgxpool.Pool
 
-	// GetSentinel returns a sentinal value of a type that's internal to River.
-	// Its purpose is to prevent public code from implementing this interface
-	// because it's not meant to be used externally at this time.
-	GetSentinel() internal.Sentinel
```

Comment:
See PR description, but I had to drop the sentinel because the two packages can no longer share an internal type (and a cyclic dependency is not allowed anyway).